index.bs

```diff
@@ -1710,3 +1738,161 @@ When [=a browsing context is discarded=], run the following steps:
  1. [=Emit an event=] with |body| and |parent id|.
 
 </div>
+
+## Log ## {#module-log}
+
+The <dfn export for=modules>log</dfn> module contains functionality and events
+related to logging.
+
+### Definition ### {#module-log-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+Command = (GetCommand)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+LogEvent = (
+  LogEntryAddedEvent
+)
+</pre>
+
+### Types ### {#module-log-types}
+
+#### Log Entry #### {#types-log-log-entry}
+
+<pre class="cddl local-cddl">
+
+LogLevel = "debug" / "info" / "warning" / "error"
+
+LogEntry = {
+  type: text,
+  level: LogLevel,
+  text: text / null,
+  timestamp: int,
+  extra: { *text => any }
+}
+
+ConsoleLogExtra = {
+  type: text,
+  realm: Realm,
+  args: [*RemoteValue],
+}
+
+JavascriptLogExtra = {
+  lineNumber: int,
+  columnNumber: int,
+  url: text
+}
+
+</pre>
+
+Each log event is represented by a <code>LogEntry</code> object. This has a
+<code>type</code> property which represents the type of log entry added, a
+<code>level</code> property representing severity, a <code>text</code> property
+with the log message string itself, and a <code>timesteamp</code> property
+corresponding to the time the log entry was generated. The <code>extra</code>
+property contains additional data specific to the given type.
+
+#### entryAdded #### {#event-log-entryAdded}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        LogEntryAddedEvent = {
+         method: "log.entryAdded",
+         params: LogEntry
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+<div algorithm="remote end event trigger for log.entryAdded">
+
+When the [=Printer=] algorithm is invoked with arguments |logLevel|, |args| and,
+optionally, <var ignore>options</var>:
+
+1. If |logLevel| is "<code>error</code>" or "<code>assert</code>", let |level|
+   be "<code>error</code>". If |logLevel| is "<code>debug</code>" or
+   "<code>trace</code>" let |level| be "<code>debug</code>". If |logLevel| is
+   "<code>warn</code>", let |level| be "<code>warning</code>". Otherwise let
+   |level| be "<code>info</code>".
+
+1. Let |timestamp| be a [=time value=] representing the current date and time in UTC.
+
+1. Let |text| be an implementation-defined string representing the arguments
```

Comment:
Oh, well, that's unfortunate. How large are the differences between implementations here? Could tests written for this event assert anything at all about what the text is? It seems like `console.log("foo")` should have the text "foo", at the very least.

Reply:
Yeah, I think asserting that string arguments are represented as themselves makes sense.